### DISCUSSION TITLE
How to reference built-in strings in built internationalization?

### DISCUSSION THREAD

**Discussion by phuctm97** [cloudscape-design/components#2640] [Q&A] [answered]
Cloudscape supports built-in internationalization https://cloudscape.design/get-started/dev-guides/internationalization

But how do I use these built-in strings in custom components?

**Comment by YueyingLu** [comment, marked as answer]
Hi @phuctm97 
thanks for reaching out,
we currently do not fully support built-in internationalization for custom components.

**Reply by phuctm97** [reply]
Should I create a PR for that? It seems like someone can just export the internal function and it's available for custom components? I can implement my own i18n but it's a waste to have duplicated strings.

**Reply by Al-Dani** [reply]
@phuctm97 this isn't a matter of implementing this functionality, but a deliberate design decision. Built-in strings were created for the specific context, we can update them because we know that they used only in that context.
For custom components we recommend to create duplicates.
